### ISSUE TITLE
Serve pensieve js from a CDN

### ISSUE THREAD

**Issue by LucianBuzzo** [resin-io/pensieve#271]
We can put the bundle pensieve JS code behind a CDN to improve load times.

**Comment by lurch** [comment]
Oh, S3 doesn't count as a CDN?

**Comment by LucianBuzzo** [comment]
@lurch S3 is a storage service and doesn't focus on content delivery, which is why AWS provide [CloudFront](https://aws.amazon.com/cloudfront/) as a CDN solution.

We should investigate publishing pensieve to npm and using https://unpkg.com for content delivery.

**Comment by lurch** [comment]
Aren't the load times more likely to be impacted by loading the actual YAML data from Github though, than they are by loading the minified JS? Although perhaps you've already profiled this... ;)

> @lurch S3 is a storage service and doesn't focus on content delivery, which is why AWS provide CloudFront as a CDN solution.

Ahhh, thanks for the info :)

**Comment by lurch** [comment]
P.S. I'm not fully aware of the "minification conventions" but should https://resin-production-downloads.s3.amazonaws.com/pensieve/latest/pensieve.js actually be named `pensieve.min.js` ?

**Comment by LucianBuzzo** [comment]
@lurch Just because load times are strongly affected by Yaml retrieval from GitHub does not make this an invalid issue. If you'd like to discuss or improve load times of GitHub data, then please feel free to open a separate issue concerning this.
Additionally, If you think that the file stored on S3 uses an incorrect naming convention then please open a separate issue for that as well.
This issue is specifically about using a content delivery network to serve the Pensieve bundle.

**Comment by lurch** [comment]
Sorry for adding in off-topic conversations. I don't feel strongly enough about any of them to warrant opening new issues for them.

**Comment by lurch** [comment]
BTW, the only reason I mentioned the GitHub YAML loading, was that I was assuming that the JS and the YAML were being loaded in parallel (from separate servers), and also assuming that the load-times for the YAML would outweigh the load-times for the JS; and so I was wondering if serving the JS from a CDN would be "optimisation effort in the wrong place". (IYSWIM)

But now that I think about it a bit more, I realise that the JS has to be loaded **first**, because it's code inside the JS that actually makes the request for the YAML :wink:  (I also didn't realise that the minified Pensieve JS is 1.5MB!)
